Symfony/CS/Fixer/Contrib/AlignEqualsFixer.php

```diff
@@ -24,49 +24,66 @@ class AlignEqualsFixer extends AbstractAlignFixer
+        // During the process of fixing internal state of single Token may be affected by injecting ALIGNABLE_PLACEHOLDER to its content.
+        // The placeholder will be resolved by `replacePlaceholder` method by removing placeholder or changing it into spaces.
+        // That way of fixing the code causes disturbances in marking Token as changed - if code is perfectly valid then placeholder
+        // still be injected and removed, which will cause the `changed` flag to be set.
+        // To handle that unwanted behavior we work on clone of Tokens collection and then override original collection with fixed collection.
+        $tokensClone = clone $tokens;
+
+        $deepestLevel = $this->injectAlignmentPlaceholders($tokensClone);
+        $content = $this->replacePlaceholder($tokensClone, $deepestLevel);
+
+        $tokens->setCode($content);
     }
 
     /**
      * Inject into the text placeholders of candidates of vertical alignment.
      *
-     * @param string $content
+     * @param Tokens $tokens
      *
-     * @return array($code, $deepestLevel)
+     * @return int $deepestLevel
      */
-    private function injectAlignmentPlaceholders($content)
+    private function injectAlignmentPlaceholders(Tokens $tokens)
     {
         $deepestLevel = 0;
-        $parenCount = 0;
-        $bracketCount = 0;
-        $tokens = Tokens::fromCode($content);
+        $limit = $tokens->count();
 
-        foreach ($tokens as $token) {
-            $tokenContent = $token->getContent();
+        for ($index = 0; $index < $limit; ++$index) {
+            $token = $tokens[$index];
 
-            if ($token->equals('=') && 0 === $parenCount && 0 === $bracketCount) {
-                $token->setContent(sprintf(self::ALIGNABLE_PLACEHOLDER, $deepestLevel).$tokenContent);
+            if ($token->equals('=')) {
+                $token->setContent(sprintf(self::ALIGNABLE_PLACEHOLDER, $deepestLevel).$token->getContent());
                 continue;
             }
 
             if ($token->isGivenKind(T_FUNCTION)) {
                 ++$deepestLevel;
-            } elseif ($token->equals('(')) {
-                ++$parenCount;
-            } elseif ($token->equals(')')) {
-                --$parenCount;
-            } elseif ($token->equals('[')) {
-                ++$bracketCount;
-            } elseif ($token->equals(']')) {
-                --$bracketCount;
+                continue;
+            }
+
+            if ($token->equals('(')) {
+                $index = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
+                continue;
+            }
+
+            if ($token->equals('[')) {
+                $index = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_INDEX_SQUARE_BRACE, $index);
+                continue;
+            }
+
+            if ($token->isGivenKind(CT_ARRAY_SQUARE_BRACE_OPEN)) {
+                $index = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_ARRAY_SQUARE_BRACE, $index);
+                continue;
```

Comment:
No, we will not. I spoke with Fabien about versioning and he want to keep it simple - only one maintained version, even no patch releases.
Due to that we may port some patches back into 1.4-dev until it will be released, but no further.

Yes, it would be nothing wrong on throwing it back into 1.5, but it context given above we will gain nothing and someone merging it (let see... probably me;) ) will need to sit down and make some regression.

Yes, I have done a little exception now.
2.x line for next major, simply to much of it to handle it in one big PR.
1.5 branch for not paralyze development, I have already asked for 1.4 release over 2 weeks ago :(